test/spec/Feature/Query/QuerySpec.hs

```diff
@@ -617,14 +622,12 @@ spec = do
           , matchHeaders = [matchContentTypeJson]
           }
 
-      it "cannot request a partitioned table as parent from a partition" $
+      -- we only search for foreign key relationships after checking the
+      -- the existence of first table
+      it "table not found error if first table does not exist" $
         get "/car_model_sales_202101?select=id,name,car_models(id,name)&order=id.asc" `shouldRespondWith`
-          [json|
-            {"hint":"Perhaps you meant 'car_model_sales' instead of 'car_model_sales_202101'.",
-             "details":"Searched for a foreign key relationship between 'car_model_sales_202101' and 'car_models' in the schema 'test', but no matches were found.",
-             "code":"PGRST200",
-             "message":"Could not find a relationship between 'car_model_sales_202101' and 'car_models' in the schema cache"} |]
-          { matchStatus  = 400
+          [json| {"code":"PGRST205","details":null,"hint":null,"message":"Could not find relation 'test.car_model_sales_202101' in the schema cache"} |]
```

Comment:
Could we add a hint that suggest a table name?. This is good when typos happen. IIRC there's already some fuzzy search library in place for this. @laurenceisla Did some previous work for this.

Reply:
> Could we add a hint that suggest a table name?. This is good when typos happen.

Sounds great!

> IRC there's already some fuzzy search library in place for this. @laurenceisla Did some previous work for this.

Oh yes,  found it: #2575